src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs

```diff
@@ -63,29 +64,39 @@ public ApplicationExtract(ILogger<ApplicationExtract> log, ApplyDataContext appl
             }
 
             _logger.LogInformation($"ApplicationExtract function executed at: {DateTime.Now}");
+            
+            var currentApplicationId = new Guid();
 
             try
             {
                 var applications = await GetApplicationsToExtract(DateTime.Now);
 
                 foreach (var applicationId in applications)
                 {
-                    var answers = await ExtractAnswersForApplication(applicationId);
+                    currentApplicationId = applicationId;
+                    try
+                    {
+                        var answers = await ExtractAnswersForApplication(applicationId);
 
-                    using (var transaction = _applyDataContext.Database.BeginTransaction())
+                        using (var transaction = _applyDataContext.Database.BeginTransaction())
+                        {
+                            await SaveExtractedAnswersForApplication(applicationId, answers);
+                            await SaveSectorDetailsForApplication(applicationId, answers);
+                            await LoadOrganisationManagementForApplication(applicationId, answers);
+                            await LoadOrganisationPersonnelForApplication(applicationId, answers);
+                            await transaction.CommitAsync();
+                        }
+                        await EnqueueApplyFilesForExtract(applyFileExtractQueue, answers);
+                    }
+                    catch (SqlException ex)
                     {
-                        await SaveExtractedAnswersForApplication(applicationId, answers);
-                        await SaveSectorDetailsForApplication(applicationId, answers);
-                        await LoadOrganisationManagementForApplication(applicationId, answers);
-                        await LoadOrganisationPersonnelForApplication(applicationId, answers);
-                        await transaction.CommitAsync();
+                        _logger.LogError(ex, "Error while processing the ApplicationExtract for application {applicationId}", applicationId);
                     }
-                    await EnqueueApplyFilesForExtract(applyFileExtractQueue, answers);
                 }
             }
             catch (Exception ex)
             {
-                _logger.LogError(ex, $"Error while processing the ApplicationExtract function at: {DateTime.Now}");
+                _logger.LogError(ex, $"Error while processing the ApplicationExtract function for application {currentApplicationId}");
```

Comment:
structured logging 
`_logger.LogError(ex, "Error while processing the ApplicationExtract function for application {applicationId}", currentApplicationId);`